src/states/elder/mod.rs

```diff
@@ -1607,27 +1597,43 @@ impl Approved for Elder {
         self.send_routing_message(RoutingMessage { src, dst, content })
     }
 
-    fn handle_relocate_event(&mut self, payload: RelocateDetails) -> Result<(), RoutingError> {
-        info!("{} - handle Relocate: {:?}.", self, payload);
-
-        if self.chain.our_prefix().matches(&payload.destination) {
-            debug!(
-                "{} - ignoring Relocate event - destination already in our section.",
-                self
-            );
+    fn handle_relocate_event(
+        &mut self,
+        details: RelocateDetails,
+        signature: BlsSignature,
+    ) -> Result<(), RoutingError> {
+        if !self.chain.can_remove_member(&details.pub_id) {
+            info!("{} - ignore Relocate: {:?} - not a member", self, details);
             return Ok(());
         }
 
-        let pub_id = payload.pub_id;
+        info!("{} - handle Relocate: {:?}.", self, details);
 
-        self.send_routing_message(RoutingMessage {
-            src: Authority::Section(self.our_prefix().name()),
-            dst: Authority::Node(*payload.pub_id.name()),
-            content: MessageContent::Relocate(payload),
-        })?;
+        let pub_id = details.pub_id;
 
-        // Delay the disconnect, to give the peer chance to receive the `Relocate` message.
-        self.remove_member(pub_id, DisconnectTime::Later);
+        let proof = {
+            let proof_for_source = self.chain.prove(&Authority::Node(*details.pub_id.name()));
+            let proof_for_target = self.chain.prove(&Authority::Section(details.destination));
+
+            if proof_for_source.blocks_len() > proof_for_target.blocks_len() {
+                proof_for_source
+            } else {
+                proof_for_target
+            }
+        };
```

Comment:
This could use a comment. I understand that we are taking the longer proof, but I'm not sure why. Is it because the longer one will be correct for both the source and the target? If yes, is it guaranteed to be so?

Reply:
In might seem it should be sufficient to only take the proof for the target section, but what can happen (at least in theory) is that the relocating node is not fully caught up to all the changes in its own section (the source section) while the target section already is. So the `proof_for_source` might actually end up being longer than `proof_for_target`. If we didn't do this, the relocating node might not be able to verify the proof.

Also, just taking the longer of the proofs should be guaranteed to be sufficient both for the source and the target.

@jeanphilippeD : could you confirm this?

Reply:
Yes, we need the longest so it can be validated by both the Relocating node and the target section.
A short comment is probably a good idea.

Reply:
OK, I'll add the comment saying something along the same lines as I wrote above you.